src/java/org/streamingpool/core/domain/StreamStreamDependencyTreeImpl.java

```diff
@@ -0,0 +1,39 @@
+package org.streamingpool.core.domain;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Multimap;
+import org.streamingpool.core.service.StreamId;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class StreamStreamDependencyTreeImpl implements StreamDependencyTree {
+
+    private final Multimap<StreamId<?>, StreamId<?>> dependencies = HashMultimap.create();
+
+    /**
+     * Add a dependency to the tree from source to parent
+     */
+    public void addDependency(StreamId<?> source, StreamId<?> parent) {
+        synchronized (dependencies) {
```

Comment:
You can maybe synchronize the whole methods to reduce the indentation.

Reply:
I tend to prefer synchronized blocks since it is easier to see than the synchronized on the method name. It also have another drawback but we can discuss this offline if you like :)

Reply:
It's not an important issue, so I think we stick with what you want. :)

Reply:
:+1: